### PR TITLE
Remove Compiler Warnings

### PR DESCRIPTION
**Pull request by yuchunc**
Also removed outdated libraries.

### CODE REVIEW

File: .formatter.exs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[
+  import_deps: [],
+  import_deps: [:phoenix],
```
Comment (alexfilatov) [inline .formatter.exs]: I believe we don’t need to import Phoenix formatting configuration in the library as uberauth is made for any plug-based elixir app, not only Phoenix.

File: lib/ueberauth/strategy/line/oauth.ex [hunk 2]
```diff
@@ -47,9 +47,10 @@ defmodule Ueberauth.Strategy.Line.OAuth do
...
   def get_token!(params \\ [], opts \\ []) do
-    token = opts
-    |> client
-    |> OAuth2.Client.get_token!(params)
+    token =
+      opts
+      |> client
+      |> OAuth2.Client.get_token!(params)
 
     token
```
Comment (alexfilatov) [inline lib/ueberauth/strategy/line/oauth.ex]: I think we can omit `token` var here
```suggestion
      opts
      |> client
      |> OAuth2.Client.get_token!(params)
```

File: mix.lock [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
-  "plug": {:hex, :plug, "1.1.6", "8927e4028433fcb859e000b9389ee9c37c80eb28378eeeea31b0273350bf668b", [:mix], [{:cowboy, "~> 1.0", [hex: :cowboy, optional: true]}]},
-  "poison": {:hex, :poison, "2.2.0", "4763b69a8a77bd77d26f477d196428b741261a761257ff1cf92753a0d4d24a63", [:mix], []},
-  "ssl_verify_fun": {:hex, :ssl_verify_fun, "1.1.0", "edee20847c42e379bf91261db474ffbe373f8acb56e9079acb6038d4e0bf414f", [:rebar, :make], []},
-  "ueberauth": {:git, "https://github.com/ueberauth/ueberauth.git", "5b34fadfdf696dd1e25f4d3317634e619dcae35e", [ref: "5b34fadfdf696dd1e25f4d3317634e619dcae35e"]}}
+%{
+  "certifi": {:hex, :certifi, "2.8.0", "d4fb0a6bb20b7c9c3643e22507e42f356ac090a1dcea9ab99e27e0376d695eba", [:rebar3], [], "hexpm", "6ac7efc1c6f8600b08d625292d4bbf584e14847ce1b6b5c44d983d273e1097ea"},
+  "dogma": {:hex, :dogma, "0.1.16", "3c1532e2f63ece4813fe900a16704b8e33264da35fdb0d8a1d05090a3022eef9", [:mix], [{:poison, ">= 2.0.0", [hex: :poison, repo: "hexpm", optional: false]}], "hexpm", "8533cb896ea527959923f9c3f08e7083e18ff681388ad7c9a599dd5d28e9085f"},
```
Comment (alexfilatov) [inline mix.lock]: .lock file still contains `dogma`

**Review by alexfilatov — APPROVED** [review]
Thank you so much, really great! 👍 
Added some notes, but not blockers.

**Comment by yuchunc** [comment]
Hey @alexfilatov 
Thanks for the speedy response! I have fixed your comments.

File: .formatter.exs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[
+  import_deps: [],
+  import_deps: [],
```
Comment (alexfilatov) [inline .formatter.exs]: I believe this line could be also removed @yuchunc

**Comment by alexfilatov** [comment]
removed line

File: .formatter.exs (hunk 4 above)
Comment (alexfilatov) [inline .formatter.exs]:
```suggestion
```

**Pull request merged**